test/System.Net.Http.Formatting.Test/HttpMessageContentTests.cs

```diff
@@ -76,6 +76,21 @@ private static async Task ValidateRequest(HttpContent content, bool containsEnti
         private static async Task ValidateResponse(HttpContent content, bool containsEntity)
         {
             Assert.Equal(ParserData.HttpResponseMediaType, content.Headers.ContentType);
+
+#if NET6_0_OR_GREATER
+            // Test initialization leaves the inner HttpContent (EmptyContent or StreamContent) w/ a non-null but
+            // unseekable Stream (EmptyReadStream). HttpMessageContent believes the message has content but is unable
+            // to calculate the message length correctly in this case. Overriding the Content e.g. w/ a StringContent
+            // makes things worse because the serialized content then contains unexpected Content-Type and
+            // Content-Length headers i.e. ParserData.HttpResponseWithEntity looks _smaller_ than
+            // ParserData.HttpResponse  though the opposite would be true in any real scenario. Buffer the
+            // HttpMessageContent to give it a better chance at calculating the length.
+            if (!containsEntity)
+            {
+                await content.LoadIntoBufferAsync();
+            }
```

Comment:
&hellip; and this is the other.

Reply:
ReadContentAsync already calls LoadIntoBufferAsync, you could just move the length check after that.

Reply:
I didn't do that because the original order looks very different to `HttpMessageContent` and I would like to keep that when testing other TFMs.